### DISCUSSION TITLE
Is there anyway to automate Apple Sign in

### DISCUSSION THREAD

**Discussion by tking16** [appium/appium#17296] [Q&A]
The app we have has multiple sign in options including Facebook, Twitter and Apple Sign in. Any ideas on how one can go about bypassing 2FA to test apple sign in? As when attempting to sign in on an iOS Simulator it prompts you to sign in with an apple account (so we can't use a fake sandbox tester here unfortunately)

**Comment by jlipps** [comment]
It looks like if you [sign into icloud](https://apple.stackexchange.com/questions/321365/does-ios-simulator-in-xcode-allow-appleid-sign-in) on a simulator (making sure to accept the TOS for the account separately in a web browser I guess), then it will allow use of sign in with apple.
